interfaces/cython/cantera/ctml_writer.py

```diff
@@ -639,12 +645,21 @@ def build(self, p):
                 nt = len(self._transport)
                 for n in range(nt):
                     self._transport[n].build(t)
+        if self._standardState:
```

Comment:
Just want to check that the behavior here is expected: this will be falsey if `_standardState` is `None`, `''`, `0`, `[]`, etc. If any of those should be allowed values, the check should instead be `if self._standardState is not None:`

Reply:
This is correct.  The accepted inputs are all strings: `unity`, `molar_volume`, and `solvent_volume`.  These are re-assigned to an int value, c.f. `IdealSolidSolnPhase::setStandardConcentrationModel`: 
```
void IdealSolidSolnPhase::setStandardConcentrationModel(const std::string& model)
{
    if (caseInsensitiveEquals(model, "unity")) {
        m_formGC = 0;
    } else if (caseInsensitiveEquals(model, "molar_volume")) {
        m_formGC = 1;
    } else if (caseInsensitiveEquals(model, "solvent_volume")) {
        m_formGC = 2;
    } else {
        throw CanteraError("IdealSolidSolnPhase::setStandardConcentrationModel",
                           "Unknown standard concentration model '{}'", model);
    }
}
```

Reply:
I think users setting strings is much better than "magic numbers", so I vote against allowing to set the integers directly :smile:

Reply:
I think this discussion is a bit confused -- the "standard state" that's being set here would be an object of type `constantIncompressible`, or nothing. The strings `molar_volume` etc. are for setting the `standard_concentration` property of the `phase` object.

Reply:
Oops - you're right.  Okay, so can you comment on @bryanwweber 's query?  It would look like it is used correctly.

If somebody enters something other than an instance of `contantIncompressible`, it looks like it repeats the build of the species thermo...?

Reply:
I agree, the `else` case for the `_standardState` makes no sense. I think it should be more like `if self._standardState:` build, otherwise do nothing.

Reply:
So skip the entire check to make sure the string assigned is actually an instance of the `standardState` class?  Just based on how other classes operate in this file, I would think it would be:

```
if self._standardState:
            ss = s.addChild("standardState")
            ss['model'] = id
            if isinstance(self._standardState, standardState):
                self._standardState.build(ss)
```

and just omit the `else` statement.  No?